### PR TITLE
feat: added intermission and hover

### PR DESCRIPTION
**Pull request by samuelOsborne**
Completes: DOTL-103

### CODE REVIEW

File: packages/dotlottie-js/src/tests/dotlottie-js-node.spec.ts [hunk 1]
```diff
@@ -165,6 +165,45 @@ describe('addAnimation', () => {
...
+      loop: true,
+      playMode: PlayMode.Bounce,
+      speed: 1.5,
+    };
+
+    dotlottie.addAnimation({
+      ...animationOptions,
+    });
+
+    dotlottie.build();
```
Comment (theashraf) [inline packages/dotlottie-js/src/tests/dotlottie-js-node.spec.ts:201]: `build` is async!!

File: packages/dotlottie-js/src/tests/dotlottie-js-browser.spec.ts [hunk 2]
```diff
@@ -165,6 +165,45 @@ describe('addAnimation', () => {
...
+      loop: true,
+      playMode: PlayMode.Bounce,
+      speed: 1.5,
+    };
+
+    dotlottie.addAnimation({
+      ...animationOptions,
+    });
+
+    dotlottie.build();
```
Comment (theashraf) [inline packages/dotlottie-js/src/tests/dotlottie-js-browser.spec.ts:201]: `build` is async!!

File: packages/dotlottie-js/src/common/lottie-animation-common.ts [hunk 3]
```diff
@@ -34,10 +34,14 @@ export class LottieAnimationCommon {
 
   private _playMode: PlayMode;
 
-  private _loop: boolean;
+  private _loop: boolean | number;
```
Comment (theashraf) [inline packages/dotlottie-js/src/common/lottie-animation-common.ts:37]: if `loop` can be either a number, maybe we need to have a validation if it's a valid non negative number.

File: packages/dotlottie-js/src/common/lottie-animation-common.ts [hunk 4]
```diff
@@ -57,6 +61,8 @@ export class LottieAnimationCommon {
     this._loop = options.loop ?? false;
     this._autoplay = options.autoplay ?? false;
     this._defaultActiveAnimation = options.defaultActiveAnimation ?? false;
+    this._hover = options.hover ?? false;
+    this._intermission = options.intermission ?? 0;
```
Comment (theashraf) [inline packages/dotlottie-js/src/common/lottie-animation-common.ts:65]: don't we need to perform runtime validation on the `intermission`, to validate if it fits in the constraints, like being a non negative number  ?

File: packages/dotlottie-js/src/tests/dotlottie-js-node.spec.ts (hunk 1 above)
Comment (samuelOsborne) [reply packages/dotlottie-js/src/tests/dotlottie-js-node.spec.ts]: added

File: packages/dotlottie-js/src/tests/dotlottie-js-browser.spec.ts (hunk 2 above)
Comment (samuelOsborne) [reply packages/dotlottie-js/src/tests/dotlottie-js-browser.spec.ts]: added

File: packages/dotlottie-js/src/common/lottie-animation-common.ts (hunk 3 above)
Comment (samuelOsborne) [reply packages/dotlottie-js/src/common/lottie-animation-common.ts:37]: added validation

**Review by theashraf — APPROVED** [review]

**Pull request merged**